backend/models/eventTemplate.model.js

```diff
@@ -1,58 +1,89 @@
+const { Location } = require('./dictionaries/location.model');
+const { TimeZone } = require('./dictionaries/timeZone.model');
 const mongoose = require('mongoose');
-
 mongoose.Promise = global.Promise;
 
 const eventTemplateSchema = mongoose.Schema({
-    name: { type: String },
-    description: { type: String },
-    type: {type: String, enum : ['Project Meeting', 'Orientation', 'Workshop', 'Hacknight']},
-    createdDate: { type: Date, default: Date.now },
-    weekDay: {type: Number, enum : [0, 1, 2, 3, 4, 5, 6]},
-    startTime: { type: Date },                          
-    endTime: { type: Date },                            
-    isActive: { type: Boolean},
-    project: {                                          
-        type: mongoose.Schema.Types.ObjectId,
-        ref: 'Project'
+  name: { type: String },
+  belongsToProjectID: { type: mongoose.Schema.Types.ObjectId }, // Active project an event is a child of, if any (structured)
+  eventManagerID: { type: mongoose.Schema.Types.ObjectId}, // ID of user that is responsible for managing event
+  locationZone: {
+    type: String,
+    validate: {
+      validator: async function (v) {
+        let loc = await Location.find({});
+        return loc[0].locations.indexOf(v) >= 0;
+      },
+      message: (props) => `${props.value} is not a valid location`,
     },
-    isOnline: { type: Boolean},
-    location: {                                         
-        addressLine1: { type: String },
-        addressLine2: { type: String },
-        city: { type: String },
-        state: { type: String },
-        zip: { type: String }
+  }, // HfLA locations project participates at (predefined in dictionary)
+  locationName: { type: String }, // Name of event location
+  description: { type: String },
+  type: { type: String, enum: ['Project Meeting', 'Orientation', 'Workshop', 'Hacknight'] },
+  createdDate: { type: Date, default: Date.now },
+  startTime: { type: Date },
+  endTime: { type: Date },
+  timeZone: {
+    type: String,
+    validate: {
+      validator: async function (v) {
+        let timeZone = await TimeZone.find({});
+        return timeZone[0].timeZones.indexOf(v) >= 0;
+      },
+      message: '{VALUE} is not a valid TimeZone',
     },
-    videoConferenceLink: { type: String }              
+  }, // Event’s time zone
+  templateID: { type: String }, // Event’s join ID connecting it to recurring instances for future editing
+  recurInterval: { type: String, enum: ['daily', 'weekly', 'bi-weekly', 'monthly'] }, // daily, weekly, bi-weekly, monthly
+  weekDay: { type: Number, enum: [0, 1, 2, 3, 4, 5, 6] }, //day of the week for the meetings -? sunday, monday, tuesday, wednesday,thrursday, friday, saturday
+  monthWeek: { type: Number, enum: [1, 2, 3, 4, 5] }, // 1st week, 2nd week, 3rd week, 4th week, 5th week - a month can have either 4 weeks or 5 weeks
+  isActive: { type: Boolean },
+  project: { type: mongoose.Schema.Types.ObjectId, ref: 'Project' },
```

Comment:
@alex-anakin -> Just curious, is "project" different from "belongsToProjectID"?

Reply:
We don't need a reference to the project here. 
But field `belongsToProjectID` emulates a foreign key.
@akibrhast please remove the `project` field from the model.